indexer/services/vulcan/src/handlers/order-replace-handler.ts

```diff
@@ -0,0 +1,393 @@
+      orderFlags !== ORDER_FLAG_SHORT_TERM &&
+      status === OrderPlaceV1_OrderPlacementStatus.ORDER_PLACEMENT_STATUS_BEST_EFFORT_OPENED
+    ) {
+      return false;
+    }
+
+    // In the case where a stateful orderPlace is opened with a more recent expiry than an
+    // existing order on the indexer, then the order will not have been placed or replaced and
+    // no subaccount message should be sent.
+    if (placeOrderResult.placed === false &&
+      placeOrderResult.replaced === false &&
+      placementStatus ===
+      OrderPlaceV1_OrderPlacementStatus.ORDER_PLACEMENT_STATUS_BEST_EFFORT_OPENED) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Removes the order from the cancelled orders cache in Redis.
+   *
+   * @param orderId
+   * @protected
+   */
+  protected async removeOrderFromCanceledOrdersCache(
+    orderId: string,
+  ): Promise<void> {
+    await runFuncWithTimingStat(
+      CanceledOrdersCache.removeOrderFromCaches(orderId, redisClient),
+      this.generateTimingStatsOptions('remove_order_from_cancel_cache'),
+    );
+  }
+
+  /**
+   * Removes and sends the cached order update for the given order id if it exists.
+   *
+   * @param orderId
+   * @returns
+   */
+  protected async sendCachedOrderUpdate(
+    orderId: string,
+    headers: IHeaders,
+  ): Promise<void> {
+    const cachedOrderUpdate: OrderUpdateV1 | undefined = await StatefulOrderUpdatesCache
+      .removeStatefulOrderUpdate(
+        orderId,
+        Date.now(),
+        redisClient,
+      );
+
+    if (cachedOrderUpdate === undefined) {
+      return;
+    }
+
+    const orderUpdateMessage: Message = {
+      key: getOrderIdHash(cachedOrderUpdate.orderId!),
+      value: Buffer.from(
+        Uint8Array.from(OffChainUpdateV1.encode({ orderUpdate: cachedOrderUpdate }).finish()),
+      ),
+      headers,
+    };
+    sendMessageWrapper(orderUpdateMessage, KafkaTopics.TO_VULCAN);
+  }
+
+  /**
+   * Updates the orderbook and sends a message to socks for the change in the orderbook.
+   * @param removeOrderResult
+   * @param perpetualMarket
+   */
+  protected async removeOldOrderFromOrderbook(
+    removeOrderResult: RemoveOrderResult,
+    perpetualMarket: PerpetualMarketFromDatabase,
+    headers: IHeaders,
+    sendWebsocketMessage: boolean,
+  ): Promise<void> {
+    const updatedQuantums: number = await runFuncWithTimingStat(
+      this.updatePriceLevelsCache(
+        removeOrderResult,
+      ),
+      this.generateTimingStatsOptions('update_price_level_cache'),
+    );
+    if (sendWebsocketMessage) {
+      const orderbookMessage: Message = {
+        value: this.createOrderbookWebsocketMessage(
+          removeOrderResult.removedOrder!,
+          perpetualMarket,
+          updatedQuantums,
+        ),
+        headers,
+      };
+      sendMessageWrapper(orderbookMessage, KafkaTopics.TO_WEBSOCKETS_ORDERBOOKS);
+    }
+  }
+
+  /**
+   * Update orderbookLevelsCache, and assumes that the order is resting on the book
```

Comment:
Given that there is a `removeOrderResult` which indicates if the order was resting on the book, is there an issue with checking the `isRestingOnBook` field vs assuming it's always true?